### PR TITLE
Don't catch the error of the heap commands when `set exception-* on`

### PR DESCRIPTION
**Pull request by lebr0nli**
Fix the nightmare when debugging the heap commands (https://github.com/pwndbg/pwndbg/pull/1242#issuecomment-1274039275)

### CODE REVIEW

**Comment by lebr0nli** [comment]
@CptGibbon, could you verify whether this fix works or not? Thanks!

File: pwndbg/commands/__init__.py [hunk 1]
```diff
@@ -276,6 +277,9 @@ def _OnlyWithResolvedHeapSyms(*a, **kw):
         e = lambda s: print(message.error(s))
         w = lambda s: print(message.warn(s))
         if pwndbg.heap.current.can_be_resolved():
+            # We don't catch the error for developers, only catch for users.
+            if gdb.parameter("exception-verbose") or gdb.parameter("exception-debugger"):
```
Comment (disconnect3d) [inline pwndbg/commands/__init__.py]: We should:
* fetch those through pwndbg.config.exception_verbose and pwndbg.config.exception_debugger
* move this to the exception case and re-raise the err `except Exception as err: if ...: raise err`

**Review by disconnect3d — CHANGES_REQUESTED** [review]
see comment

File: pwndbg/commands/__init__.py (hunk 1 above)
Comment (lebr0nli) [reply pwndbg/commands/__init__.py]: Ok, I'll fix it with your suggestions!

**Comment by CptGibbon** [comment]
I'll take a look today, thanks for changing this.

**Comment by CptGibbon** [comment]
This does indeed print out a backtrace when `set exception-verbose on` is used 👍

File: pwndbg/commands/__init__.py [hunk 2]
```diff
@@ -276,21 +277,26 @@ def _OnlyWithResolvedHeapSyms(*a, **kw):
...
             except Exception as err:
-                e(f"{function.__name__}: Unknown error: `{err}` when running this symbols.")
+                e(f"{function.__name__}: An unknown error occurred when running this command.")
                 if pwndbg.config.resolve_heap_via_heuristic:
                     w(
                         "Maybe you can try to determine the libc symbols addresses manually, set them appropriately and re-run this command. For this, see the `heap_config` command output and set the `main_arena`, `mp_`, `global_max_fast`, `tcache` and `thread_arena` addresses."
                     )
                 else:
                     w("You can try `set resolve-heap-via-heuristic on` and re-run this command.\n")
+                if pwndbg.config.exception_verbose.value or pwndbg.config.exception_debugger.value:
```
Comment (disconnect3d) [inline pwndbg/commands/__init__.py]:
```suggestion
                if pwndbg.config.exception_verbose or pwndbg.config.exception_debugger:
```

Comment (disconnect3d) [reply pwndbg/commands/__init__.py]: The `pwndbg.config.Parameter` class has a `__bool__` magic method/operator defined:
```py
    def __bool__(self):
        return bool(self.value)
```

so we can omit `.value` here :)

**Review by disconnect3d — APPROVED** [review]

**Pull request merged**